gap/examples.gi

```diff
@@ -370,3 +370,66 @@ GeneralisedPetersenGraphCons);
 
 InstallMethod(GeneralisedPetersenGraph, "for integer, integer", [IsInt, IsInt],
 {n, k} -> GeneralisedPetersenGraphCons(IsImmutableDigraph, n, k));
+
+InstallMethod(HaarGraphCons,
+"for IsMutableDigraph and one integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  local m, binaryList, D, i, j;
+  if not IsPosInt(n) then
+    ErrorNoReturn("the argument <n> must be a positive integer,");
+  fi;
+
+  m := Log(n, 2) + 1;
+  binaryList := DIGRAPHS_BlistNumber(n + 1, m);
+
+  D := EmptyDigraph(IsMutableDigraph, 2 * m);
+
+  for i in [1 .. m] do
+    for j in [1 .. m] do
+      if binaryList[((j - i) mod m) + 1] then
+          DigraphAddEdge(D, [i, m + j]);
+      fi;
+    od;
+  od;
+
+  return DigraphSymmetricClosure(D);
+end);
+
+InstallMethod(HaarGraphCons,
+"for IsImmutableDigraph, integer, int",
+[IsImmutableDigraph, IsInt],
+function(filt, n)
+  local D;
+  D := MakeImmutable(HaarGraphCons(IsMutableDigraph, n));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(HaarGraph, "for a function, integer",
+[IsFunction, IsInt],
+HaarGraphCons);
+
+InstallMethod(HaarGraph, "for integer", [IsInt],
+{n} -> HaarGraphCons(IsImmutableDigraph, n));
+
+# TODO: Taken from semigroups...
```

Comment:
Shouldn't leave TODOs in the code, though I'm not entirely sure what the best way of resolving this is. Either we should change `SEMIGROUPS_BlistNumber` to `DIGRAPHS_BlistNumber` everywhere in Semigroups or just make it an alias of `DIGRAPHS_BlistNumber`

Reply:
Ok, I will ask about it today...